Meadowlark-js/tests/e2e/Locations.test.ts

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { getAccessToken, Clients } from './functions/Credentials';
+import { createResource, deleteResourceByLocation } from './functions/Resources';
+import { createSchool } from './functions/Shared';
+import { baseURLRequest, rootURLRequest } from './Setup';
+
+describe('Locations', () => {
+  describe('with strict validation', () => {
+    it('should fail when missing required properties', async () => {
+      await baseURLRequest
+        .post('/v3.3b/ed-fi/locations')
+        .auth(await getAccessToken(Clients.Vendor1), { type: 'bearer' })
+        .send({
+          classroomIdentificationCode: 'string',
+          schoolReference: {
+            schoolId: 99,
+          },
+          maximumNumberOfSeats: 20,
+          optimalNumberOfSeats: 10,
+        })
+        .expect(400)
+        .then((response) => {
+          expect(response.body.message).toContain('Resource School is missing identity');
+        });
+    });
+
+    describe('when school is added', () => {
+      let schoolId: number;
+      let schoolLocation: string;
+      let location: string;
+
+      beforeAll(async () => {
+        schoolId = 100;
+        schoolLocation = await createSchool(schoolId);
+      });
+
+      it('should add location with valid school', async () => {
+        location = await createResource({
+          endpoint: 'locations',
+          body: {
+            classroomIdentificationCode: 'string',
+            schoolReference: {
+              schoolId,
+            },
+            maximumNumberOfSeats: 20,
+            optimalNumberOfSeats: 10,
+          },
+        });
+
+        await rootURLRequest
+          .get(location)
+          .auth(await getAccessToken(Clients.Vendor1), { type: 'bearer' })
+          .expect(200);
```

Comment:
Shouldn't this be 201?

Reply:
you're right. I will review this one

Reply:
This is the request to verify that the location is correct after the resource was created, this should be a 200

Reply:
OK that makes sense now. Shouldn't there be an assert for 201 response code?

Reply:
Line 44 calls the method createResource, this method verifies the 201 and returns the location